File: proxy/src/main/java/com/example/proxy/app/v2/OrderServiceV2.java
package com.example.proxy.app.v2;

import com.example.proxy.app.v1.OrderRepositoryV1;

public class OrderServiceV2 {

    private final OrderRepositoryV2 orderRepositoryV2;

    public OrderServiceV2(OrderRepositoryV2 orderRepositoryV2) {
        this.orderRepositoryV2 = orderRepositoryV2;
    }

    public void order(String itemId) {
        orderRepositoryV2.save(itemId);
    }

}

File: advanced/src/main/kotlin/com/example/advanced/app/v2/OrderServiceV2.kt
package com.example.advanced.app.v2

import com.example.advanced.trace.TraceId
import com.example.advanced.trace.hellotrace.HelloTraceV2
import org.springframework.stereotype.Service


@Service
class OrderServiceV2(
    private val orderRepositoryV2: OrderRepositoryV2,
    private val trace: HelloTraceV2
) {

    fun order(traceId: TraceId, itemId: String) {
        val status = trace.beginSync(traceId,"OrderService.order")
        try {
            orderRepositoryV2.save(status.traceId,itemId)
            trace.end(status)
        } catch (e: Exception) {
            trace.exception(status, e)
            throw e
        }
    }
}
